lib/transformer/caster/template.rb

```diff
@@ -0,0 +1,28 @@
+module Transformer
+  module Caster
+    class Template
```

Comment:
@antw This is a class where I have my doubts over. It does do it's job - in a minimal sense - but I don't like how the code turned out. Especially the `add_method` and `add_area` split up. Am I mingling two purposes into a single class or?